src/wp-includes/rewrite.php

```diff
@@ -158,7 +158,7 @@ function add_rewrite_rule( $regex, $query, $after = 'bottom' ) {
  */
 function add_rewrite_tag( $tag, $regex, $query = '' ) {
 	// validate the tag's name
-	if ( strlen( $tag ) < 3 || $tag[0] != '%' || $tag[ strlen( $tag ) - 1 ] != '%' ) {
+	if ( strlen( $tag ) < 3 || '%' != $tag[0] || '%' != $tag[ strlen( $tag ) - 1 ] ) {
```

Comment:
This would be the right Yoda expression: `3 > strlen( $tag )`

Reply:
Because `$tag` is wrapped in a function call it cannot accidentally be assigned to a value to either way around is valid.